diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -308,13 +309,18 @@ fun ASTNode?.isAccessibleOutside(): Boolean =
 
 fun ASTNode.hasSuppress(warningName: String): Boolean {
     return parent({ node ->
-        node.findChildByType(MODIFIER_LIST)
-                ?.findAllNodesWithSpecificType(ANNOTATION_ENTRY)
+        val children: ASTNode? = if (node.elementType != FILE) {
+            node.findChildByType(MODIFIER_LIST)
+        } else {
+            node.findChildByType(FILE_ANNOTATION_LIST)
+        }
+        children?.findAllNodesWithSpecificType(ANNOTATION_ENTRY)
+                ?.map { it.psi as KtAnnotationEntry }
                 ?.any {
-                    (it.psi as KtAnnotationEntry).shortName.toString() == Suppress::class.simpleName
-                            && (it.psi as KtAnnotationEntry).valueArgumentList?.arguments?.isNotEmpty() ?: true
-                            && (it.psi as KtAnnotationEntry).valueArgumentList?.arguments
-                            ?.get(0)?.text?.contains(warningName) ?: true
+                    it.shortName.toString() == Suppress::class.simpleName
+                            && it.valueArgumentList?.arguments?.isNotEmpty() ?: false // don't suppress anything
+                            && it.valueArgumentList?.arguments
+                            ?.firstOrNull()?.text?.contains(warningName) ?: true
```

Comment:
```suggestion
                            && it.valueArgumentList?.arguments
                            ?.firstOrNull()?.text?.contains(warningName) ?: false
```
If list is empty, `firstOrNull` will return null and result will be `false`.